public/js/deleteCategory.js

```diff
@@ -0,0 +1,21 @@
+const deleteButtons = document.querySelectorAll('.deleteCatBtn');
+deleteButtons.forEach(deleteCatBtn => {
+    deleteCatBtn.addEventListener('click', async (e) => {
+        const id = e.target.dataset.id;
+
+        try {
+            const response = await fetch('/api/deleteCategory', {
+                method: 'POST',
+                body: JSON.stringify({"id": id}),
+                headers: {
+                    "Content-Type": "application/json"
+                }
+            });
+            const responseData = await response.json();
+            document.cookie = `response = ${responseData.message}`;
+            window.location.reload();
```

Comment:
not a fan of this but it is consistent with the rest of the app (unfortunately). however you need to check the response is successful before reloading